mitodl/ocw-data-parser
# Strip whitespace from external links

#### What are the relevant tickets?
Closes #71 

#### What's this PR do?
Strips whitespace from links in `course_foreign_files`

#### How should this be manually tested?
Run ocw-data-parser on a course mentioned in the issue, you should not get upload errors about the specified links.
- Create a sample json file: 
```
{
    "courses": [
      "cms-611j-creating-video-games-fall-2014"
      ]
}
```

Download the raw files for it and generate master JSON:
```python
from ocw_data_parser import OCWDownloader
from ocw_data_parser import OCWParser

downloader = OCWDownloader("spaces.json", "PROD", "raw", "ocw-content-storage")
downloader.download_courses()

parser = OCWParser("raw/cms-611j-creating-video-games-fall-2014/", "output")
parser.generate_master_json()
```

The `course_foreign_files` section will be at the end, make sure there are no blank spaces at the end of any of the links.
